runtime/parachains/src/paras/benchmarking.rs

```diff
@@ -29,17 +29,31 @@ fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
 	assert_eq!(event, &system_event);
 }
 
+fn generate_disordered_pruning<T: Config>() {
+	let mut needs_pruning = Vec::new();
+
+	for i in 0 .. 1000 {
+		let id = ParaId::from(i);
+		let block_number = T::BlockNumber::from(1000u32);
+		needs_pruning.push((id, block_number));
+	}
+
+	<Pallet<T> as Store>::PastCodePruning::put(needs_pruning);
+}
+
 benchmarks! {
 	force_set_current_code {
-		let c in 1024 .. 2048;
+		let c in 1 .. MAX_CODE_SIZE;
 		let new_code = ValidationCode(vec![0; c as usize]);
 		let para_id = ParaId::from(c as u32);
+		generate_disordered_pruning::<T>();
 	}: _(RawOrigin::Root, para_id, new_code)
 	verify {
 		assert_last_event::<T>(Event::CurrentCodeUpdated(para_id).into());
 	}
 	force_set_current_head {
-		let new_head = HeadData(vec![0]);
+		let max_head_data_size = configuration::Pallet::<T>::config().max_head_data_size as usize;
```

Comment:
Not sure this is going to work because this is a storage item, and can be modified on chain.

We need something like a configurable constant which also sets the upper limit for what this storage value can be.

Does that make sense?